spec/support/matchers/email_matchers.rb

```diff
@@ -0,0 +1,22 @@
+RSpec::Matchers.define :receive_email do
+  match do |address|
+    emails =
+      ActionMailer::Base.deliveries.select { |e| e.to.include?(address) }
+
+    emails.select! { |e| e.subject.match(@subject) } if @subject
+    emails.select! { |e| e.text_part.body.match(@body) } if @body
+    emails.any?
+  end
+
+  chain :with_subject do |subject|
```

Comment:
Don't think they don't get referenced as instance variables in the `match` block with the shorthand approach.